generators/pythonv2/codegen/src/ast/Type.ts

```diff
@@ -0,0 +1,195 @@
+    type: "str";
+}
+
+interface Bytes {
+    type: "bytes";
+}
+
+interface List {
+    type: "list";
+    value: Type;
+}
+
+interface Set {
+    type: "set";
+    value: Type;
+}
+
+interface Tuple {
+    type: "tuple";
+    values: Type[];
+}
+
+interface Dict {
+    type: "dict";
+    keyType: Type;
+    valueType: Type;
+}
+
+interface None {
+    type: "none";
+}
+
+interface Optional {
+    type: "optional";
+    value: Type;
+}
+
+interface Union {
+    type: "union";
+    values: Type[];
+}
+
+interface Any {
+    type: "any";
+}
+
+export class Type extends AstNode {
+    private internalType: InternalType;
+
+    private constructor(internalType: InternalType) {
+        super();
+        this.internalType = internalType;
+    }
+
+    public static int(): Type {
+        return new Type({ type: "int" });
+    }
+
+    public static float(): Type {
+        return new Type({ type: "float" });
+    }
+
+    public static bool(): Type {
+        return new Type({ type: "bool" });
+    }
+
+    public static str(): Type {
+        return new Type({ type: "str" });
+    }
+
+    public static bytes(): Type {
+        return new Type({ type: "bytes" });
+    }
+
+    public static list(value: Type): Type {
+        return new Type({ type: "list", value });
+    }
+
+    public static set(value: Type): Type {
+        return new Type({ type: "set", value });
+    }
+
+    public static tuple(values: Type[]): Type {
+        return new Type({ type: "tuple", values });
+    }
+
+    public static dict(keyType: Type, valueType: Type): Type {
+        return new Type({ type: "dict", keyType, valueType });
+    }
+
+    public static none(): Type {
+        return new Type({ type: "none" });
+    }
+
+    public static optional(value: Type): Type {
+        return new Type({ type: "optional", value });
+    }
+
+    public static union(values: Type[]): Type {
+        return new Type({ type: "union", values });
+    }
+
+    public static any(): Type {
+        return new Type({ type: "any" });
+    }
+
+    public write(writer: Writer): void {
+        switch (this.internalType.type) {
+            case "int":
+                writer.write("int");
+                break;
+            case "float":
+                writer.write("float");
+                break;
+            case "bool":
+                writer.write("bool");
+                break;
+            case "str":
+                writer.write("str");
+                break;
+            case "bytes":
+                writer.write("bytes");
+                break;
+            case "list":
+                writer.write("List[");
+                this.internalType.value.write(writer);
+                writer.write("]");
+                break;
+            case "set":
+                writer.write("Set[");
+                this.internalType.value.write(writer);
+                writer.write("]");
+                break;
+            case "tuple":
+                writer.write("Tuple[");
```

Comment:
@noanflaherty you will probably want to setup `ClassReference` and imports in this PR since we want to make sure that we are importing appropriately from the `typing` module 

`from typing import Tuple`
`from typing import Set`